support/docker/production/docker-compose.yml

```diff
@@ -49,29 +39,23 @@ services:
     restart: "always"
 
   postgres:
-    image: postgres:10-alpine
+    image: postgres:12-alpine
```

Comment:
Hello,  
Be careful, updating postgres could mean manual manipulation, have you tested the upgrade?

Reply:
I am testing it at the moment.

Reply:
Migrating beetween major postgres version requires some manual dump/restore, but otherwise works:

```
docker-compose -f support/docker/production/docker-compose.yml exec -T postgres pg_dumpall -U postgres > pgdump
```

And after updating:

```
docker-compose -f support/docker/production/docker-compose.yml exec -i postgres psql -U postgres < pgdump
```

Restart the postgres service and delete `pgdump`.

Reply:
It may seems obvious but I'd add that before running the postgres service again and execute the restore command, we need to delete all db files in the docker-volume to re-init the db after restarting the service : 
```bash
rm -rf ./docker-volume/db
```
Otherwise we got this error even after the restore command : 
```
postgres_1       | 
postgres_1       | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres_1       | 
postgres_1       | 2020-03-26 19:21:19.547 UTC [1] FATAL:  database files are incompatible with server
postgres_1       | 2020-03-26 19:21:19.547 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 10, which is not compatible with this version 12.2.
```

Reply:
> Migrating beetween major postgres version requires some manual dump/restore, but otherwise works. Execute on the running peertube service before updating the image:
> 
> ```
> docker-compose -f support/docker/production/docker-compose.yml exec -T postgres pg_dumpall -U postgres > pgdump
> ```
> 
> And after updating the image, run the postgres service and execute the restore command:
> 
> ```
> docker-compose -f support/docker/production/docker-compose.yml exec -i postgres psql -U postgres < pgdump
> ```
> 
> Restart the postgres service and delete `pgdump`.

You actually don't need a `pg_dumpall` because restarting the postgres service will recreate roles and peertube database template according to the docker-compose setup.

Dump command : 
> docker-compose exec -T postgres_service pg_dump -U postgres_user postgres_db > pgdump

Restore command (after deleting `./docker-volume/db` mounted database and `compose up postgres_service`):
> docker-compose exec -T postgres_service pg_restore -U postgres_user -d postgres_db < pgdump